content/advent-2016/saga-go-dependency-management.md

```diff
@@ -0,0 +1,106 @@
++++
+date = "2016-12-13T00:00:00"
+author = [ "Sam Boyer (@sdboyer)" ]
+title = "The Saga of Go Dependency Management"
+series = ["Advent 2016"]
++++
+
+# The Saga of Go Dependency Management
+
+The Go community is on the cusp of some major shifts in the way we handle dependencies. These shifts are a long time coming, and involve the work and effort of dozens, if not hundreds, of people. Six to twelve months from now, the Go dependency management landscape is likely to look very different.
+
+For those who haven't been following closely - or, honestly, even for those who have - it can be difficult to keep track of what's going on, and where things are going. So, as part of the Gopher advent series, we thought we'd tie up Go package management past, present, and future in a nice big bow.
+
+## The road so far
+
+Anyone who takes a stroll down the memory lane of Go package management will find ample clumps of ripped-out hair strewn about. The process has been long and frustrating, and there's far too much to cover in detail here. But the history is important to where we are now, so a high-level timeline is worth it.
+
+`go get` was, and is, the only official tooling for retrieving Go code and placing it on disk. It has served Go users and developers alike since the release of Go 1.0. Hand-in-hand with `go get` has been the general recommendation from the Go Team on how to keep everything working well: "write backwards-compatible code." That's a slightly-rephrased version of the Go 1 guarantee. This pairs well with `go get`, as `go get`'s behavior could only be correct if the following were true:
```

Comment:
Maybe instead of:

> Hand-in-hand with `go get` has been the general recommendation from the Go Team on how to keep everything working well: "write backwards-compatible code."

Try:

> Since `go get` could fetch any public version of a dependency, the Go team had an additional recommendation: "write backwards-compatible code."